lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1689,6 +1689,8 @@ def main():
                 # has to be poweredoff first
                 pyv.set_powerstate(vm, 'poweredoff', module.params['force'])
             result = pyv.remove_vm(vm)
+        elif module.params['snapshot_op']:
```

Comment:
verify length of snapshot_op dict instead of __not_None__ else you will trigger snapshot_op on non absent condition